client/consensus/babe/src/lib.rs

```diff
@@ -502,6 +543,16 @@ impl<B, C, E, I, Error, SO> sc_consensus_slots::SimpleSlotWorker<B> for BabeWork
 		s
 	}
 
+	fn notify_slot(
+		&self,
+		_parent_header: &B::Header,
+		slot_number: SlotNumber,
+		epoch_descriptor: &ViableEpochDescriptor<B::Hash, NumberFor<B>, Epoch>,
+	) {
+		self.slot_notification_sinks.lock()
+			.retain_mut(|sink| sink.try_send((slot_number, epoch_descriptor.clone())).is_ok());
```

Comment:
The docs never explicitly mentions it, but from my readings of the source code (https://docs.rs/futures-channel/0.2.0/src/futures_channel/mpsc/mod.rs.html#416-429), `try_send` will never block and returns error if the buffer is full.